ext/opentelemetry-ext-boto/src/opentelemetry/ext/boto/__init__.py

```diff
@@ -0,0 +1,271 @@
+            }
+            if region_name:
+                meta[REGION] = region_name
+
+            for key, value in meta.items():
+                span.set_attribute(key, value)
+
+            # Original func returns a boto.connection.HTTPResponse object
+            result = original_func(*args, **kwargs)
+            span.set_attribute("http.status_code", getattr(result, "status"))
+            span.set_attribute("http.method", getattr(result, "_method"))
+
+            return result
+
+    def _patched_auth_request(self, original_func, instance, args, kwargs):
+        operation_name = None
+
+        frame = currentframe().f_back
+        operation_name = None
+        while frame:
+            if frame.f_code.co_name == "make_request":
+                operation_name = frame.f_back.f_code.co_name
+                break
+            frame = frame.f_back
+
+        endpoint_name = getattr(instance, "host").split(".")[0]
+
+        with self._tracer.start_as_current_span(
+            "{}.command".format(endpoint_name), kind=SpanKind.CONSUMER,
+        ) as span:
+            if args:
+                http_method = args[0]
+                span.resource = "%s.%s" % (endpoint_name, http_method.lower())
+            else:
+                span.resource = endpoint_name
+
+            add_span_arg_tags(
+                span,
+                endpoint_name,
+                args,
+                AWS_AUTH_ARGS_NAME,
+                AWS_AUTH_TRACED_ARGS,
+            )
+
+            # Obtaining region name
+            region_name = _get_instance_region_name(instance)
+
+            meta = {
+                AGENT: "boto",
+                OPERATION: operation_name,
+            }
+            if region_name:
+                meta[REGION] = region_name
+
+            for key, value in meta.items():
+                span.set_attribute(key, value)
+
+            # Original func returns a boto.connection.HTTPResponse object
+            result = original_func(*args, **kwargs)
+            span.set_attribute("http.status_code", getattr(result, "status"))
+            span.set_attribute("http.method", getattr(result, "_method"))
+
+            return result
+
+
+def truncate_arg_value(value, max_len=1024):
+    """Truncate values which are bytes and greater than ``max_len``.
+    Useful for parameters like "Body" in ``put_object`` operations.
+    """
+    if isinstance(value, bytes) and len(value) > max_len:
+        return b"..."
+
+    return value
+
+
+def add_span_arg_tags(span, endpoint_name, args, args_names, args_traced):
+    if endpoint_name not in ["kms", "sts"]:
+        tags = dict(
+            (name, value)
+            for (name, value) in zip(args_names, args)
+            if name in args_traced
+        )
+        tags = flatten_dict(tags)
+        for key, value in {
+            k: truncate_arg_value(v)
+            for k, v in tags.items()
+            if k not in {"s3": ["params.Body"]}.get(endpoint_name, [])
```

Comment:
should that s3 dict be S3_BLACKLIST_TAGS? looks identical.

Reply:
actually looking at this code, does it make sense? getting the endpoint_name field from a static dictionary, it'll never return anything besides an empty list.

Reply:
Removed `S3_BLACKLIST_TAGS`. I think that code makes sense if `endpoint_name` can be `s3`.